infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -333,14 +346,33 @@ private void startDiscoveryService(ServerContext serverContext) {
         // Start LogReplicationDiscovery Service, responsible for
         // acquiring lock, retrieving Site Manager Info and processing this info
         // so this node is initialized as Source (sender) or Sink (receiver)
-        replicationDiscoveryService = new CorfuReplicationDiscoveryService(serverContext, runtime);
+        replicationDiscoveryService = new CorfuReplicationDiscoveryService(serverContext, runtime, clusterManagerAdapter);
         replicationDiscoveryService.start();
     }
 
+    /**
+     * Create the Cluster Manager Adapter, i.e., the adapter to external provider of the topology.
+     *
+     * @return cluster manager adapter instance
+     */
+    private CorfuReplicationClusterManagerAdapter getClusterManagerAdapter(String pluginConfig) {
+        LogReplicationPluginConfig config = new LogReplicationPluginConfig(pluginConfig);
```

Comment:
In the previous PR (https://github.com/CorfuDB/CorfuDB/commit/12228fb8b4c6b429c8ed05191b233682dce36c80) we have tried to consume an existing object for LogReplicationPluginConfig rather than creating a new object. Why the same is not implemented here?

Reply:
in the previous PR, the pluginConfig instance was saved in the replicationContext, and other classes referred to the pluginConfig via the replication context. The replication context comes much later in our code path.
This particular class is the entry point to LR, and doesn't have access to the replicationContext. 

And since the LogReplicationPluginConfig is light weight, I think its okay to have an instance here.

Reply:
Sounds good to me.